### PR TITLE
[pickers] Inherit previous and next icons size from their parent button

### LINKED ISSUE #8206: [pickers] Can't change the size of the previous / next icons
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Summary 💡

I would like to be able to control the size of the next / previous buttons in the StaticDateRangePicker date range calendar selection.

image

right now these are forced to `fontSize` of "small"

https://github.com/mui/mui-x/blob/master/packages/x-date-pickers/src/internals/components/PickersArrowSwitcher/PickersArrowSwitcher.tsx#L111

https://github.com/mui/mui-x/blob/master/packages/x-date-pickers/src/internals/components/PickersArrowSwitcher/PickersArrowSwitcher.tsx#L128

### Examples 🌈

https://github.com/mui/mui-x/blob/master/packages/x-date-pickers/src/internals/components/PickersArrowSwitcher/PickersArrowSwitcher.tsx#L111

https://github.com/mui/mui-x/blob/master/packages/x-date-pickers/src/internals/components/PickersArrowSwitcher/PickersArrowSwitcher.tsx#L128

### Motivation 🔦

I have a custom date picker component where I use the standard StaticDateRangePicker w/ no modifications beyond the spec, with the "small" right icon showing (example here)

image

and a custom component to the right using a icon w/ a medium font size (example here)

image

I've also indicated the icon DOM for the icon in the date range component.

The differences in sizes is causing a layout issue and I don't have a way to control the padding of the date range icon.

### PR DESCRIPTION
**Pull request by flaviendelangle**
Fixes #8206

### CODE REVIEW

File: packages/x-date-pickers/src/internals/components/PickersArrowSwitcher/PickersArrowSwitcher.tsx [hunk 1]
```diff
@@ -108,7 +108,7 @@ export const PickersArrowSwitcher = React.forwardRef(function PickersArrowSwitch
     elementType: PreviousIconButton,
     externalSlotProps: slotProps?.previousIconButton ?? componentsProps?.previousIconButton,
     additionalProps: {
-      size: 'small',
+      size: 'medium',
```
Comment (flaviendelangle) [inline packages/x-date-pickers/src/internals/components/PickersArrowSwitcher/PickersArrowSwitcher.tsx:111]: The size of the icon rendered was equivalent to a `medium`
Do we want smaller icons or do we want to keep the current UI ?

Comment (LukasTy) [reply packages/x-date-pickers/src/internals/components/PickersArrowSwitcher/PickersArrowSwitcher.tsx:111]: The current UI is `small` (34px), the proposed `medium` produces `40px`. 🤷‍♂️ 🙈 
I'm not sure we should be making this change at this moment... 🤔

Comment (flaviendelangle) [reply packages/x-date-pickers/src/internals/components/PickersArrowSwitcher/PickersArrowSwitcher.tsx:111]: On my computer, to keep the same icon size, I need to put `medium`

Comment (LukasTy) [reply packages/x-date-pickers/src/internals/components/PickersArrowSwitcher/PickersArrowSwitcher.tsx:111]: Ah, sorry. I might have looked at a cached version or just didn't have straight eyes.
Visually it's the same. 👌 
It's just the fact that now the height "requested by header content" is even higher (40px) than previously (34px), even though we force the max-height to `30px`.
Shouldn't be a problem now, but might cause certain issues down the line.
**Current:**
Screenshot 2023-03-14 at 11 12 30
**This implementation:**
Screenshot 2023-03-14 at 11 12 47

**Review by LukasTy — APPROVED** [review]

File: packages/x-date-pickers/src/internals/components/PickersArrowSwitcher/PickersArrowSwitcher.tsx (hunk 1 above)
Comment (flaviendelangle) [reply packages/x-date-pickers/src/internals/components/PickersArrowSwitcher/PickersArrowSwitcher.tsx:111]: Indeed
Maybe @gerdadesign would have an opinion here between using a "medium" `IconButton` or a "small" `IconButton` for those icons by default.

**Pull request merged**